### DISCUSSION TITLE
Autolock not working

### DISCUSSION THREAD

**Discussion by paulrudy** [blacs30/bitwarden-alfred-workflow#112] [Q&A]
Autolock isn't working for me. I've given `LOCK_TIMEOUT` a value of 60, and successfully installed autolock with the keyword.

**Comment by blacs30** [comment]
It might take more time, up to 65 minutes for it to work. The agent runs every 5 minutes and checks the last usage.
The lock only happens if you haven't used Bitwarden for at least 60 minutes.

Can you run this command in a terminal to check that it doesn't throw any error exit codes, 0 is expected:

```
launchctl list | ag liso

SHOULD RETURN:

-       0       com.lisowski-development.alfred.bitwarden_lock
```

**Comment by luckman212** [comment]
@paulrudy just a note: `ag` (silver searcher) doesn't come installed by default... so try `launchctl list | grep liso` if @blacs30's command doesn't work

**Comment by blacs30** [comment]
@luckman212 of course. completely overlooked that. thanks for that.

**Comment by paulrudy** [comment]
Thanks. It has not locked itself in over 24 hours. When I run `launchctl list | grep liso`, it returns:
```
-	0	com.lisowski-development.alfred.bitwarden_lock
-	127	com.lisowski-development.alfred.bitwarden_autosync
```

**Comment by blacs30** [comment]
@paulrudy sorry for my late reply.
Error 127 happens when a binary/executable is not found but that is regarding the sync daemon, you could try to reinstall it.

Regarding the auto lock you could debug it the following way:

```
file=$(cat ~/Library/LaunchAgents/bw_auto_lock.sh_lock_agent.plist | grep Users | sed -n 's:.*<string>\(.*\)</string>.*:\1:p')
sed '/last-usage ].*/i set -x' "${file}" > "${file}".tmp && mv "${file}".tmp "${file}" && chmod +x "${file}"
bash "${file}" 1
```

This will get the `bw_auto_lock.sh` in the Alfred directory from the LaunchAgents plist config.
Then it adds a `set -x` close to the end where it compares the last-usage timestamp (written by the Alfred workflow into the data dir of the workflow) with the current timestamp 
And at last the file is executed with a lock timeout of 1 minute and you can see more output, which you also could post here in case it's still not working.

**Reply by paulrudy** [reply]
Thanks @blacs30. I reinstalled the sync daemon, that worked. For auto lock, I ran the commands you posted above and got:

```
sed: 1: "/last-usage ].*/i set -x": command i expects \ followed by text
found workflow v2.4.0 at /Users/paul/Library/Mobile Documents/com~apple~CloudDocs/Alfred/Alfred.alfredpreferences/workflows/user.workflow.0FBF9871-6599-49CE-812B-09C0B31CA6C4
```

Still not auto locking. Any ideas?

**Comment by kaspersoerensen** [comment]
I also have some problems with auto-lock. After system reboot, the vault is not locked. I have set the value of auto-lock to 120.

```
launchctl list | grep liso
-	126	com.lisowski-development.alfred.bitwarden_lock
```

Reinstalling auto-lock does not seem to fix the issue.
